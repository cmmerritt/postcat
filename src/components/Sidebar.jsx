import React from 'react';
import SidebarItem from './SidebarItem.jsx';

const Sidebar = ({ history }) => {
  console.log(history, 'sidebar history');
  const sidebarElements = history.map((h, i) => 
  { console.log(h);
    return (
      <li key={i}>
        <SidebarItem 
          {...h}
        />
      </li>
    );
  });

  return (
    <ul>
      {sidebarElements}
    </ul>
  );
};

export default Sidebar;
