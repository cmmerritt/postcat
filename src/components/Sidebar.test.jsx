import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Sidebar from './Sidebar';

describe('Sidebar component', () => {
  afterEach(() => cleanup());
  it('renders Sidebar', () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
