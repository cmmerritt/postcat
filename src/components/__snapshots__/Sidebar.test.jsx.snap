// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders Sidebar 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
